modules/solid_mechanics/contrib/neml2_moose/neml2/base/config.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023, UChicago Argonne, LLC
+// All Rights Reserved
+// Software Name: NEML2 -- the New Engineering material Model Library, version 2
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <cstddef>
+#include <torch/torch.h>
+
+// clang-format off
+#define NEML2_DTYPE torch::kFloat64
+
+#define NEML2_INT_DTYPE torch::kInt64
+// clang-format on
+
+/**
+ * TODO: make the following constants configurable
+ */
+/// The machine precision
+#define NEML2_EPS 1E-15
+/// The tolerance used in various algorithms
+#define NEML2_TOL 1E-6
+/// A tighter tolerance used in various algorithms
+#define NEML2_TOL2 1e-12
```

Comment:
those should be adjusted if `NEML2_DTYPE` get changed to `torch::kFloat32` for example, right?

Reply:
That's correct. We just haven't fully thought through how best to handle all the constants yet. They might all be run-time constants in the future.